src/code/z_skin.c

```diff
@@ -1,21 +1,222 @@
+        } else {
+            Struct_800A598C_2* phi_s0 = &temp_s3[0];
+            f32 phi_f26 = 0.0f;
+            f32 phi_f24 = 0.0f;
+            f32 phi_f22 = 0.0f;
+            while (phi_s0 < &temp_s3[temp_a0_2]) {
+                Vec3f sp88;
+                f32 temp_f20 = phi_s0->unk_8 * 0.01f;
+                sp88.x = phi_s0->x;
+                sp88.y = phi_s0->y;
+                sp88.z = phi_s0->z;
+                SkinMatrix_Vec3fMtxFMultXYZ(&D_801600C0[phi_s0->unk_0], &sp88, &spD0);
+                spD0.x *= temp_f20;
+                spD0.y *= temp_f20;
+                spD0.z *= temp_f20;
+                phi_f26 += spD0.x;
+                phi_f24 += spD0.y;
+                phi_f22 += spD0.z;
+                phi_s0++;
+            }
+            spDC.x = phi_f26;
+            spDC.y = phi_f24;
+            spDC.z = phi_f22;
+        }
+        func_800A57C0(&D_801600C0[temp_s3[phi_s6->unk_4].unk_0], temp_v0, phi_s6, spEC, &spDC);
+        phi_s6++;
+    }
+    gSPSegment(POLY_OPA_DISP++, 0x08, avb->buf[avb->unk_0]);
+    if (avb->unk_0 == 0) {
+        avb->unk_0 = 1;
+    } else {
+        avb->unk_0 = 0;
+    }
+
+    CLOSE_DISPS(gfxCtx, "../z_skin.c", 344);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_800A598C.s")
+#endif
+
+void func_800A5E28(GraphicsContext* gfxCtx, PSkinAwb* skin, s32 limbIndex, s32 arg3, s32 arg4) {
+    SkinLimb** limbs;
+    Struct_800A5E28* temp_t9;
+
+    OPEN_DISPS(gfxCtx, "../z_skin.c", 364);
+    limbs = (SkinLimb**)SEGMENTED_TO_VIRTUAL(skin->skeletonHeader->segment);
+    temp_t9 = (Struct_800A5E28*)SEGMENTED_TO_VIRTUAL(((SkinLimb*)SEGMENTED_TO_VIRTUAL(limbs[limbIndex]))->segment);
+    if (!(arg4 & 1)) {
+        func_800A598C(gfxCtx, skin, limbIndex, arg3);
+    }
+    gSPDisplayList(POLY_OPA_DISP++, temp_t9->unk_8);
+    CLOSE_DISPS(gfxCtx, "../z_skin.c", 377);
+}
+
+void func_800A5F60(GraphicsContext* gfxCtx, PSkinAwb* skin, s32 limbIndex, Gfx* arg3, s32 arg4) {
+    Gfx* gfx = arg3;
+    SkinLimb** limbs;
+    s32 pad;
+
+    OPEN_DISPS(gfxCtx, "../z_skin.c", 395);
+    limbs = SEGMENTED_TO_VIRTUAL(skin->skeletonHeader->segment);
+    if (arg3 == NULL) {
+        gfx = (Gfx*)((SkinLimb*)SEGMENTED_TO_VIRTUAL(limbs[limbIndex]))->segment;
+    }
+    if (gfx != NULL) {
+        Mtx* mtx = SkinMatrix_MtxFToNewMtx(gfxCtx, &D_801600C0[limbIndex]);
+        if (mtx != NULL) {
+            gSPMatrix(POLY_OPA_DISP++, mtx, G_MTX_PUSH | G_MTX_MUL | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, gfx);
+            gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
+            gDPPipeSync(POLY_OPA_DISP++);
+        }
+    }
+    CLOSE_DISPS(gfxCtx, "../z_skin.c", 433);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin/func_800A5E28.s")
+void func_800A60D8(Actor* actor, GlobalContext* globalCtx, PSkinAwb* skin, SkinCallback callback, SkinCallback2 arg4,
+                   s32 arg5, s32 arg6, s32 arg7) {
```

Comment:
these callbacks look to be the equivalent of override and post limb draw. not documenting now is fine but just wanna point that out